client/me/purchases/manage-purchase/purchase-meta.jsx

```diff
@@ -39,443 +39,459 @@ import {
+	const translate = useTranslate();
+	const moment = useLocalizedMoment();
 
-	static defaultProps = {
-		hasLoadedSites: false,
-		hasLoadedUserPurchasesFromServer: false,
-		purchaseId: false,
-		getManagePurchaseUrlFor: managePurchase,
-	};
+	const purchase = useSelector( ( state ) => getByPurchaseId( state, purchaseId ) );
+	const site = useSelector( ( state ) => getSite( state, purchase?.siteId ) ) || null;
+	const owner = useSelector( ( state ) => getUser( state, purchase?.userId ) ) || null;
 
-	renderPrice() {
-		const { purchase, translate } = this.props;
-		const { priceText, currencyCode, productSlug } = purchase;
-		const plan = getPlan( productSlug ) || getProductFromSlug( productSlug );
-		let period = translate( 'year' );
-
-		if ( isOneTimePurchase( purchase ) || isDomainTransfer( purchase ) ) {
-			return translate( '%(priceText)s %(currencyCode)s {{period}}(one-time){{/period}}', {
-				args: { priceText, currencyCode },
-				components: {
-					period: <span className="manage-purchase__time-period" />,
-				},
-			} );
-		}
+	const isDataLoading = useSelector( isRequestingSites ) || ! hasLoadedPurchasesFromServer;
 
-		if ( isIncludedWithPlan( purchase ) ) {
-			return translate( 'Free with Plan' );
-		}
+	if ( isDataLoading || ! purchaseId ) {
+		return <PurchaseMetaPlaceholder />;
+	}
 
-		if ( plan && plan.term ) {
-			switch ( plan.term ) {
-				case TERM_BIENNIALLY:
-					period = translate( 'two years' );
-					break;
+	return (
+		<>
+			<ul className="manage-purchase__meta">
+				<PurchaseMetaOwner translate={ translate } owner={ owner } />
+				<li>
+					<em className="manage-purchase__detail-label">{ translate( 'Price' ) }</em>
+					<span className="manage-purchase__detail">
+						<PurchaseMetaPrice purchase={ purchase } translate={ translate } />
+					</span>
+				</li>
+				<PurchaseMetaExpiration
+					purchase={ purchase }
+					site={ site }
+					siteSlug={ siteSlug }
+					translate={ translate }
+					moment={ moment }
+					getEditPaymentMethodUrlFor={ getEditPaymentMethodUrlFor }
+					getManagePurchaseUrlFor={ getManagePurchaseUrlFor }
+				/>
+				<PurchaseMetaPaymentDetails
+					purchase={ purchase }
+					translate={ translate }
+					getEditPaymentMethodUrlFor={ getEditPaymentMethodUrlFor }
+					siteSlug={ siteSlug }
+					site={ site }
+					moment={ moment }
+				/>
+			</ul>
+			<RenewErrorMessage purchase={ purchase } translate={ translate } site={ site } />
+		</>
+	);
+}
 
-				case TERM_MONTHLY:
-					period = translate( 'month' );
-					break;
-			}
+PurchaseMeta.propTypes = {
+	hasLoadedPurchasesFromServer: PropTypes.bool.isRequired,
+	purchaseId: PropTypes.oneOfType( [ PropTypes.number, PropTypes.bool ] ).isRequired,
+	siteSlug: PropTypes.string.isRequired,
+	getManagePurchaseUrlFor: PropTypes.func,
+	getEditPaymentMethodUrlFor: PropTypes.func,
+};
+
+PurchaseMeta.defaultProps = {
+	hasLoadedPurchasesFromServer: false,
+	purchaseId: false,
+	getManagePurchaseUrlFor: managePurchase,
```

Comment:
While I believe `defaultProps` does still work for functional components, using regular object deconstruction default values is more idiomatic and doesn't require knowledge of React's special syntax.

That is, you could do this:

```js
export default function PurchaseMeta( {
	purchaseId = false,
	hasLoadedPurchasesFromServer = false,
	siteSlug,
	getEditPaymentMethodUrlFor,
	getManagePurchaseUrlFor = managePurchase,
} ) {
```